Neos.ContentRepository.Core/Classes/ContentRepository.php

```diff
@@ -247,6 +244,19 @@ public function getContentGraph(WorkspaceName $workspaceName): ContentGraphInter
         return $this->projectionState(ContentGraphFinder::class)->getByWorkspaceName($workspaceName);
     }
 
+    public function findWorkspaceByName(WorkspaceName $workspaceName): ?Workspace
+    {
+        return $this->getWorkspaceFinder()->findOneByName($workspaceName);
+    }
+
+    public function getWorkspaces(): Workspaces
```

Comment:
todo discuss `find` vs `get` here. The philosophy to use get was we dont find - eg filter - for anything.
But when we want to use `find` to signal the the implementation is probably slow (at least uncached) then we should go with that.

Reply:
I would suggest not to use `find` = `slow` and `get` = `fast` as a _general_ concept, because it tries to change the meaning of those words.
In this case we don't "find" any workspaces, we just get them all – at least that was my original reasoning :)

However we decide, some doc comment would be nice
```suggestion
    /**
     * Returns all workspaces of this content repository. To limit the set, {@see Workspaces::find()} and {@see Workspaces::filter()} can be used
     */
    public function getWorkspaces(): Workspaces
```

Reply:
iirc our convention is
* get for what is already there (like in-memory computations, property access)
* find for everything that has to be fetched from somewhere

Or in short: yes, we decided to go for get* for fast and find* for slow stuff and I'd rather stick to that.

Reply:
Im torn in between. `getContentGraph` for example also does one SQL query to get the currentContentStreamId, though its "get" there ... but im fine with either way.

Reply:
I couldnt find discuss post to our find* vs get* philosophy but i found a few old artefacts regarding why `Node::getParent` was renamed to `Node::findParent` in the old Neos world:

> [TODO: rename to findParent() because it is a DB operation](https://github.com/skurfuerst/contentrepository-development-collection/blob/b5e3d80782ff43e686a9001cc7a8f1e82a0dcf32/Neos.EventSourcedContentRepository/Classes/Domain/Projection/Content/TraversableNode.php#L65-L66)

and here https://github.com/neos/neos-development-collection/pull/2202

> adjust TraversableNodeInterface to have proper method namings. Everything which queries something should be called find*; as opposed to get*.

or here https://github.com/neos/neos-development-collection/pull/2202#discussion_r233797157

> `$node->findNodePath()` sounds weird to me. Why not just `getNodePath()`?

> Because we say: "all operations which are slow should be named find*", vs. "all operations which operate on already-loaded data should be named get*". And fetching a node path is a potentially costly operation.

And in `findParent` we also definitely dont filter anything still it continues to be named `Subgraph::findParentNode`.

Though there is one exception `Subgraph::findNodePath` was eventually renamed to `Subgraph::retrieveNodePath` with https://github.com/neos/neos-development-collection/pull/4090
So it seems `retrieveParent` or `retrieveWorkspaces` would also be possibly valid when starting out fresh.

But for now for consistency i guess its a good idea to stick the 6year old philosophy and name it `findWorkspaces` instead (especially we could consider adding a filter later:)).

Reply:
Just to avoid confusion: My point was that this kind of "magic meaning" makes sense for a single API, Content(Sub)Graph in this case. I would suggest to adapt in in _general_ if it leads to weird method names.
But I'm fine with either solution in this case!!

Reply:
I really find the distinction between `find*` and `get*` hard to grasp, and keep track of. Internals might and can change such that the meaning of find/get is no longer correct, and we cannot change the public API according to our internals. I think this warrants discussion, as I struggled with this naming before as well...